src/main/scala/cromwell/engine/backend/jes/JesRuntimeInfo.scala

```diff
@@ -27,10 +20,33 @@ object JesRuntimeInfo {
     new Resources()
       .setRamGb(runtimeAttributes.memoryGB)
       .setCpu(runtimeAttributes.cpu)
-      .setPreemptible(runtimeAttributes.preemptible)
       .setZones(runtimeAttributes.zones.asJava)
       .setDisks(runtimeAttributes.disks.asJava)
   }
 }
 
-case class JesRuntimeInfo(resources: Resources, docker: DockerExecutor)
+object NonPreemptibleJesRuntimeInfo extends JesRuntimeInfoBuilder {
+  def apply(commandLine: String, runtimeAttributes: CromwellRuntimeAttributes): JesRuntimeInfo = {
+    /*
+     It should be impossible for docker to be None here. Enforcing that w/ ADTs seemed more trouble than
+      it was worth. If you're ever debugging a NoSuchElementException which leads you here, that means
+      the more trouble than worth calculation was incorrect and we should have separate RuntimeAttributes for
+      docker and no docker cases
+    */
+    val dockerImage = runtimeAttributes.docker.get
+    val resources = buildResources(runtimeAttributes).setPreemptible(false)
+    new NonPreemptibleJesRuntimeInfo(resources, buildDockerExecutor(commandLine, dockerImage))
+  }
+}
+
+object PreemptibleJesRuntimeInfo extends JesRuntimeInfoBuilder {
+  def apply(commandLine: String, runtimeAttributes: CromwellRuntimeAttributes): JesRuntimeInfo = {
+    // See comment above
+    val dockerImage = runtimeAttributes.docker.get
+    val resources = buildResources(runtimeAttributes).setPreemptible(true)
+    new PreemptibleJesRuntimeInfo(resources, buildDockerExecutor(commandLine, dockerImage))
+  }
+}
```

Comment:
No, I totally agree w/ you - although we likely disagree on the resolution.

IMO the way to go is to make the apply on the companion object of the base trait which then creates and returns the correct one.